src/arch/arm/machine/gic_v2.c

```diff
@@ -93,6 +94,7 @@ BOOT_CODE static void dist_init(void)
     /* level-triggered, 1-N */
     for (i = 64; i < nirqs; i += 32) {
         gic_dist->config[i >> 5] = 0x55555555;
+        gic_dist_config_cache[i >> 5] = gic_dist->config[i >> 5];
```

Comment:
This could just be my lack of understanding, but is this by itself guaranteed to copy all of config to the cache. IMO it would be safer to do something like the below at the end of the function

```
for (int i = 0; i < 32; i++) {
      gic_dist_config_cache[i].= gic_dist->config[i]
}
``